Sources/TuistCore/Graph/GraphLoader.swift

```diff
@@ -317,10 +297,17 @@ public final class GraphLoader: GraphLoading {
         var xcframeworks: [AbsolutePath: GraphDependency] = [:]
         var packages: [AbsolutePath: [String: Package]] = [:]
 
-        init(projects: [Project]) {
+        init(projects: [Project]) throws {
             let allProjects = Dictionary(uniqueKeysWithValues: projects.map { ($0.path, $0) })
-            let allTargets = allProjects.mapValues {
-                Dictionary(uniqueKeysWithValues: $0.targets.map { ($0.name, $0) })
+            let allTargets = try allProjects.mapValues { project -> [String: Target] in
+                let projectTargetsKeysWithValues = project.targets.map { ($0.name, $0) }
+                let duplicates = projectTargetsKeysWithValues
```

Comment:
Why isn't this logic part of the linter? If it can't be in the linter for reason X, I recommend leaving a comment here explaining the why. It'll be easier for someone reviewing the code in the future to understand the rationale behind it.

Reply:
Yeah, let's move it to the linter, I initially considered to have it hear to avoid doing the same loop twice, yet I don't think it's that costly. I will move it to the linter it makes more sense :)

Reply:
```
    public func lintProject(at path: AbsolutePath, projects: [Project]) throws {
        try lintTargetDuplicationProjects(projects)
        let cache = GraphLoader.Cache(projects: projects)
        let cycleDetector = GraphCircularDetector()
        try lintProject(path: path, cache: cache, cycleDetector: cycleDetector)
    }

    // MARK: - Private
    
    private func lintTargetDuplicationProjects(_ projects: [Project]) throws {
        try projects.forEach { project in
            let duplicates = project.targets.map { $0.name }.spm_findDuplicates()
            guard duplicates.isEmpty else {
                throw GraphLoadingError.targetDuplicated(duplicates)
            }
        }
    }
```
This is how it will look, it need to be before the cache creation because `Dictionary(uniqueKeysWithValues` will crash otherwise. What do you think @adellibovi ? In the case of keeping here is more linked to the crash, while there it's more close to the lint error.

Reply:
Update, I think it even makes more sense and easier to have it into the `ManifesterLinter`, duplication of target names is local to a project manifest only and there we have the information ready for us, plus the code will be easier.